main-app/app/scripts/bingoNumbers/callingNumbers.js

```diff
@@ -0,0 +1,45 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.LongPollingService')
+        .service('longPolling',['$timeout', 'proxy', 'userModel', 'numberService', function ($timeout, proxy, userModel, numberService) {
+            var me = this;
+            me.currentNumber = 0;
+            me.startNumber = 1;
+            me.winnerMessage ='';
+            me.numbersToWin = 5;
+            me.winnerHasNotBeenFound = true;
+
+            me.startPolling = function() {
+                proxy.getBingoNumber(userModel.token, userModel.name, userModel.currentBalance, me.startNumber).then(function(response) {
+                    me.currentNumber = response.payload.call;
+                    me.startNumber += 1;
+                    checkForWinnerfound(response);
+                    //TODO: Check for number to colour in and find out how many numbers are left to win.
+                    if(me.winnerHasNotBeenFound){
+                        $timeout(me.startPolling, 2000);
+                    }
+                });
+            };
+
+            var checkForWinnerfound = function (data) {
+                if(data.message === "Line"){
+                    numberService.linePrize = false;
+                    lineFound(data);
+                }
+                else if (data.message === 'Winner') {
+                    fullHouseFound(data);
+                }
+            };
+
+            var fullHouseFound = function (data) {
+                me.winnerHasNotBeenFound = false;
+                me.winnerMessage = "Full house! you have won £" + data.payload.winnerInfo.houseprize;
```

Comment:
This is embedding UI inside the a service which simplifies communication with the server. This breaks the lasagne-ness of the architecture.